test/src/test/java/org/corfudb/runtime/view/ManagementViewTest.java

```diff
@@ -1664,4 +1658,97 @@ public void unblockSequencerRecoveryOnDeadlock() throws Exception {
         getManagementServer(SERVERS.PORT_0).getManagementAgent().triggerSequencerBootstrap(layout)
                 .get();
     }
+
+    /**
+     * Scenario to verify that we do not regress the token count when the layout switches primary
+     * sequencers.
+     * We assert that the failover sequencer should always receive a full bootstrap message rather
+     * than an empty bootstrap message (without streamTailsMap)
+     */
+    @Test
+    public void failoverSeqDoesNotRegressTokenValue() throws Exception {
+        getManagementTestLayout();
+        CorfuRuntime corfuRuntime = getDefaultRuntime();
+
+        // Append data.
+        IStreamView streamView = corfuRuntime.getStreamsView()
+                .get(CorfuRuntime.getStreamID("testStream"));
+        final byte[] payload = "test".getBytes();
+        final int num = 10;
+        // 0 - 9
+        for (int i = 0; i < num; i++) {
+            streamView.append(payload);
+        }
+
+        // Adding rule to partition out PORT_0.
+        addServerRule(SERVERS.PORT_0, new TestRule().always().drop());
+
```

Comment:
should we wait for PORT_0 to show as unresponsive?

Reply:
The next statement is the append. The append takes care of this and will only proceed if PORT_0 is removed from the log unit chain. So we achieve this indirectly.

Reply:
ok, perfect!